### ISSUE TITLE
Error for GoF test when par/par2 are too close to boundaries

### ISSUE THREAD

**Issue by tnagler** [tnagler/VineCopula#21]
`BiCopGofTest` throws errors when the parameters are too close to the boundaries. There are two separate issues:
1. When Kendall's tau is positive (but close to zero) some of the bootstrap samples may have negative Kendall's tau. Many of the implemented families do not allow for positive and negative tau and, thus, the function throws an error.
2. For the t copula, the gradient is approximated numerically by comparing the derivatives at  `par2 +- 1e3`. When `par2` is close to 2, this causes an error. 

A reproducible example:

```
library("VineCopula")
set.seed(1)

## some bivariate copulas
cop1 <- BiCop(3, tau = 0.05)  # tau close to 0
cop2 <- BiCop(4, tau = 0.05)  # tau close to 0
cop3 <- BiCop(2, tau = 0.1, par2 = 2.01)   # df close to 2

## simualte data
sim1 <- BiCopSim(100, cop1)
sim2 <- BiCopSim(100, cop2)
sim3 <- BiCopSim(100, cop3)

## GoF test
BiCopGofTest(sim1[, 1], sim1[, 2], 3)
#> Error: 
#>  In BiCopCheckTaus: Clayton copula cannot be used for tau<=0. 
BiCopGofTest(sim2[, 1], sim2[, 2], 4)
#> Error: 
#>  In BiCopCheckTaus: Gumbel copula cannot be used for tau<0.  
BiCopGofTest(sim3[, 1], sim3[, 2], 2)
#> Error: 
#>  In BiCopDeriv: The degrees of freedom parameter of the t-copula 
#>  has to be larger than 2.
```

@ulf85: I don't want to mess with your functions. Could you look into that? For the Kendall's tau issue, the solution may require changing something globally in the preprocessors. If so, please check with me before committing.

**Comment by thonghuynguyen** [comment]
I used data simulated from your example code above

##> estimate par
parCop <- BiCopEst(sim1[, 1], sim1[, 2],family=7)

##> GoF test
Gof<-BiCopGofTest(sim1[, 1], sim1[, 2],par=parCop$par,par2=parCop$par2,family=parCop$family,method="kendall",B=1000)
#>Not Error but got many of this"
#>"The BB1 or survival BB1 copula cannot be used for negatively dependent data."

##> correlation is positive
cor(sim1[, 1], sim1[, 2],method="kendall")
#>[1] 0.01939394

**Comment by ulf85** [comment]
Regarding the problem with par2 near 2 for the t-copula, we disabled the parameter check for the sub-function used in `BiCopGofTest`.
For data with a really smal Kendall's tau I suggest to use the independence test first and not the generall goodness-of-fit test.
Nevertheless, I made a workaround for the bootstrap cases where the test ends up with an error due to not appropriate Kendall's tau. These bootstraps were deleted (set to NA) and thereby not considered for the calculation of the p-value. Note that this may result in an erroneous p-value, since the empirical distribution of the test statistics may be shifted.
An appropriate warning is now printed in the output of the function.
